[16.0][MIG] fleet vehicle purchase : Migration to 16.0

lmarion-source opened the pull request.

etobella committed:
[ADD] fleet_vehicle_purchase

Commit:
[UPD] Update fleet_vehicle_purchase.pot

OCA-git-bot committed:
[BOT] post-merge updates

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (9 of 9 strings)

Translation: fleet-14.0/fleet-14.0-fleet_vehicle_purchase
Translate-URL: https://translation.odoo-community.org/projects/fleet-14-0/fleet-14-0-fleet_vehicle_purchase/es/

marcelsavegnago committed:
[FIX] fleet_vehicle_purchase: fix prepare_account_move_line method

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (9 of 9 strings)

Translation: fleet-14.0/fleet-14.0-fleet_vehicle_purchase
Translate-URL: https://translation.odoo-community.org/projects/fleet-14-0/fleet-14-0-fleet_vehicle_purchase/it/

marcelsavegnago committed:
Added translation using Weblate (Portuguese (Brazil))

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (9 of 9 strings)

Translation: fleet-14.0/fleet-14.0-fleet_vehicle_purchase
Translate-URL: https://translation.odoo-community.org/projects/fleet-14-0/fleet-14-0-fleet_vehicle_purchase/pt_BR/

lmarion-source committed:
[IMP] fleet_vehicle_purchase: black, isort, prettier

lmarion-source committed:
[16.0][MIG] fleet_vehicle_purchase : Migration to 16.0

acsonefho commented on fleet_vehicle_purchase/models/fleet_vehicle.py:
```diff
@@ -0,0 +1,31 @@
+
+    @api.depends("purchase_order_ids")
+    def _compute_purchase_order_count(self):
+        for record in self:
```
The best way to do it is with a `read_group(...)`. Something like:
```
data = self.env["purchase.order"].read_group(
    [('id', 'in', self.ids)],
    ["fleet_vehicle_id"],
    ["fleet_vehicle_id"],
)
mapped_data = {
    x["fleet_vehicle_id"][0]: x["fleet_vehicle_id_count"] for x in data
}
for record in self:
    record.purchase_order_count = mapped_data.get(record.id, 0)
```

acsonefho commented on fleet_vehicle_purchase/models/purchase_order.py:
```diff
@@ -0,0 +1,43 @@
+    fleet_vehicle_id = fields.Many2one("fleet.vehicle", string="Vehicle")
+
+
+class PurchaseOrderLine(models.Model):
```
Should be splited into a new file.

acsonefho commented on fleet_vehicle_purchase/models/purchase_order.py:
```diff
@@ -0,0 +1,43 @@
+    @api.depends("fleet_vehicle_from_po")
+    def _compute_fleet_vehicle_id(self):
+        for rec in self:
+            if rec.fleet_vehicle_from_po:
```
I don't understand this boolean field. It sets to `True` when the order is related to a fleet vehicle. So this if section will always be `True` in this case.
So when the order is not related to a fleet vehicle, it'll fill the `fleet_vehicle_id` is an empty recordset (and that's also the expected behavior).

lmarion-source replied:
This is to cover a practical case where we want to order pieces for several fleet vehicle in one PO:

- If we put the vehicle on the PO, it will be propagate on all purchase order lines and the vehicle will be readonly in the view. In this case the PO is for one vehicle.
- If we don't put the vehicle on the PO, we can order pieces for different vehicles in one PO, changing the vehicle on each line. This allows a big customer to order pieces for more than one vehicle in one PO.

lmarion-source committed:
[IMP] Add the possibility to have more than 1 vehicle by PO
   Initially, there was only one vehicle for one PO.
   Now, if the vehicle is set on the PO, all the PO lines are related to this vehicle
   But if the vehicle is not set on the PO, the user can decide to have different vehicles on every line.
   Use case is a grouped purchase of several items for several vehicles.

ivantodorovich commented:
/ocabot migration fleet_vehicle_purchase

marcelsavegnago commented:
ping @kaynnan @WesleyOliveira98

marcelsavegnago commented:
@lmarion-source can you rebase please ?

WesleyOliveira98 reviewed (approved):
LGTM

ivantodorovich reviewed (approved):
/ocabot merge nobump
